Init/nex 877/remote git repository init (#4)
oat-sa/extension-remote-proctoring: README.md

# extension tao-remote-proctoring

> This extension provide functionality to connect Tao instance to Proctorio tool

Reviewer: please provide full extended readme

<!--
EXTENSION_NAME = remoteProctoring, e.g. taoFooBar
REPOSITORY_NAME = extension-remote-proctoring, e.g. extension-tao-foo-bar
-->

# TAO _Remote Proctoring_ extension

![TAO Logo](https://github.com/oat-sa/taohub-developer-guide/raw/master/resources/tao-logo.png)

> ![GitHub](https://img.shields.io/github/license/oat-sa/extension-remote-proctoring.svg)

Reviewer: for me this does not displays when looking at the reamme.mn

Author: For me, it's not fully working right now. But I think this will work after development. I can remove it. 

```suggestion
```

![GitHub commit activity](https://img.shields.io/github/commit-activity/y/oat-sa/extension-remote-proctoring.svg)

## Table of contents
 - [Installation](#installation-instructions)
 - [Configuration](##Configuration)

> This extension provide functionality to connect Tao instance to Proctorio tool

> Using this extension u can use [Proctorio](https://proctorio.com) tool with TAO

Reviewer: i think this is ok, this extension is currently generated for Proctorio but it will contain more providers
also 
```suggestion
Using this extension you can use the remote proctoring provider [Proctorio](https://proctorio.com) tool with TAO
```

## Installation instructions

These instructions assume that you have already a TAO installation on your system. If you don't, go to
[package/tao](https://github.com/oat-sa/package-tao) and follow the installation instructions.

If you installed your TAO instance through [package-tao](https://github.com/oat-sa/package-tao),

> `oat-sa/extension-remote-proctoring` is very likely already installed. You can verify this under _Settings -> Extension

Reviewer: maybe it is ok here to specify the VCS

> manager_, where it would appear on the left hand side as `{{EXTENSION_NAME}}`. Alternatively you would find it in

Reviewer: ```suggestion
manager_, where it would appear on the left hand side as `RemoteProctoring`. Alternatively you would find it in
```

the code at `/config/generis/installation.conf.php`.

_Note, that you have to be logged in as System Administrator to do this._

Add the extension to your TAO composer and to the autoloader:
```bash
composer require oat-sa/extension-remote-proctoring
```

Install the extension on the CLI from the project root:

**Linux:**
```bash

> sudo php tao/scripts/installExtension oat-sa/extension-remote-proctoring

Reviewer: i think this uses the folder name so something like
```suggestion
sudo php tao/scripts/installExtension 'remoteProctoring'
```

Author: I think rather: 
```suggestion
sudo php tao/scripts/installExtension oat-sa/remote-proctoring
```

Reviewer: https://github.com/oat-sa/extension-remote-proctoring/blob/9350fd8eabbb5ea4f53d7360876e1e7616308e4f/composer.json#L2

```

**Windows:**
```bash

> php tao\scripts\installExtension oat-sa/extension-remote-proctoring

Reviewer: please see before comments 
installExtension  uses extension folder name (name)

Author: ```suggestion
php tao\scripts\installExtension oat-sa/remote-proctoring
```

Reviewer: --> https://hub.taotesting.com/articles/how-to-make-a-new-extension-3-3/how-to-make-a-new-extension-in-tao-33
https://github.com/oat-sa/extension-remote-proctoring/pull/4/files#diff-096f34b8841df0fda7fda714e51755f8L26

according to this 2 docs please use my suggestion

```

As a system administrator you can also install it through the TAO Extension Manager:
- Settings (the gears on the right hand side of the menu) -> Extension manager

> - Select _{{EXTENSION_NAME}}_ on the right hand side, check the box and hit _install_

Reviewer: ```suggestion
- Select remoteProctoring on the right hand side, check the box and hit _install_
```

[package/tao](https://github.com/oat-sa/package-tao) and follow the installation instructions.

_Note, that you have to be logged in as System Administrator to do this._

Add the extension to your TAO composer and to the autoloEader:
```bash
composer require oat-sa/extension-remote-proctoring
```

__Warning!__ Extension is under private oat project and it's not published in packagist. 
Please remember to add custom vcs to your `composer.json` file

```
{
  "type": "vcs",
  "url": "git@github.com:oat-sa/extension-remote-proctoring.git"
}
```

Install the extension on the CLI from the project root:

**Linux:**
```bash
sudo php tao/scripts/installExtension.php remoteProctoring
```

**Windows:**
```bash
php tao\scripts\installExtension remote-proctoring
```

As a system administrator you can also install it through the TAO Extension Manager:
- Settings (the gears on the right hand side of the menu) -> Extension manager
- Select _remoteProctoring_ on the right hand side, check the box and hit _install_

<!-- Not all of the blocks below are applicable for any repository, please remove those that aren't -->

> ## Configuration

Reviewer: maybe we can put the seed file example from @jbout 
```
      "remoteProctoring": {
        "ProctorioApiService": {
          "type": "configurableService",
          "class": "oat\\remoteProctoring\\model\\ProctorioApiService",
          "options": {
            "persistence": "default_kv",
            "oauthKey": "here goes the authKey",
            "oauthSecret": "here goes the authSecret"
          }
        }
      }
```

Author: It's not there yet. I would add this after this service will be implemented.
